src/vs/editor/contrib/rtv/RTVSynthDisplay.ts

```diff
@@ -303,48 +397,389 @@ export class RTVSynthDisplayBox implements IRTVDisplayBox{
+			const sel = window.getSelection()!;
+			let offset = sel.anchorOffset;
+			const range = document.createRange();
+
+			let isString = cell.innerText[0] === '\'' || cell.innerText[0] === '"';
+
+			let dest: HTMLElement = cell;
+			while (dest.firstChild && (!dest.classList.contains('monaco-tokenized-source') || dest.childNodes.length == 1)) {
+				dest = dest.firstChild as HTMLElement;
+			}
+
+			if (dest.childNodes.length > 1) {
+				let isNegNum = dest.firstChild!.textContent == '-';
+				offset = isNegNum ? cell.innerText.length : dest.childNodes.length - 1;
 			} else {
-				this._cellDictionary[name] = [cell];
+				// We need to carefully pick the offset based on the type
+
+				// Select the actual text
+				while (dest.firstChild) {
+					dest = dest.firstChild as HTMLElement;
+				}
+
+				offset = isString ? cell.innerText.length - 1 : cell.innerText.length;
+			}
+
+			if (this._cursorPos!.node.id === cell.id) {
+				try {
+					range.selectNodeContents(dest);
+					range.setStart(dest, offset);
+					range.collapse(true);
+
+					sel.removeAllRanges();
+					sel.addRange(range);
+				} catch (e) {
+					// TODO Better error handling
+					console.error(e);
+					range.selectNodeContents(dest);
+					range.setStart(dest, 0);
+					range.collapse(true);
+
+					sel.removeAllRanges();
+					sel.addRange(range);
+				}
+			}
+			else {
+				// console.error(`cursorPos: ${this._cursorPos!.node.id}; currCell: ${cell.id}`);
+				this.select(this._cursorPos!.node);
 			}
 		}
+	}
 
+	private async toggleIfChanged(
+		env: any,
+		varname: string,
+		cell: HTMLElement,
+		updateBoxContent: boolean = true
+	): Promise<boolean> {
+		// Keep track of changes
+		let success = false;
+		if (cell) {
+			const currentValue = cell.textContent!;
+			if (env[varname] !== currentValue) {
+				success = await this.toggleElement(env, cell, varname, true, updateBoxContent);
+			} else {
+				success = true;
+			}
+		} else {
+			console.error('toggleIfChanged called, but parent can\' be found: ');
+			console.error(cell);
+		}
+		return success;
+	}
 
-		// // Remove any existing content
-		// cell.childNodes.forEach((child) => cell.removeChild(child));
+	private async toggleElement(
+		env: any,
+		cell: HTMLElement,
+		varname: string,
+		force?: boolean,
+		updateSynthBox: boolean = true
+	): Promise<boolean> {
+		let time = env['time'];
+		let row = this.findParentRow(cell);
+		let on: boolean;
+
+		if (!time) {
+			on = true;
+		} else if (force !== undefined) {
+			on = force;
+		} else {
+			on = !this._includedTimes!.has(time);
+		}
 
-		// if (editable) {
-		// 	// make the TableCellElement `td` editable if applicable
-		// 	cell.contentEditable = 'true';
-		// }
+		if (on) {
+			let error = await this.requestValidateInput!(cell.textContent!);
+			if (error) {
+				this.addError(error, cell, 500);
+				return false;
+			}
 
-		// // Add the new content
-		// cell.appendChild(cellContent);
+			env[varname] = cell.innerText.trim();
```

Comment:
The view shouldn't modify the model directly (or at all really). From what I understand, the view should only tell the controller about user interactions. Then it's up to the controller to modify the state according to those interactions, and update/redraw the view with the latest model.